### PR TITLE
changed the workflow

### PR DESCRIPTION
**Pull request by maxwai**
changed so that the workflow on pull requests is not done twice when pushing to a branch that already has a pull request open.

Also changed that containers are only build on pushes to main as pull requests from forks don't have access to the secrets needed to push a container

### CODE REVIEW

**Comment by schitcrafter** [comment]
If I have a pull request from a fork to this repo, and push more commits to the base branch, will this run the action again? Because I don't think it will, push is only activated for branches on this repo while pull_request opened/reopened is only activated when the pr is, well, opened or reopened.

**Comment by maxwai** [comment]
> Because I don't think it will, push is only activated for branches on this repo while pull_request opened/reopened is only activated when the pr is, well, opened or reopened.

You're right, I need to change that.

**Comment by schitcrafter** [comment]
One possibility might be push only on main, and then all pull requests. That would include everything on main + all branches, extern or not, that have an active pr. only possibility of executing twice is from a pr with base main, which shouldn't happen. (copied from [poem](https://github.com/poem-web/poem/blob/7c9cff63afd54da62d637d49a91d7eb806b4a446/.github/workflows/ci.yml#L3-L7

**Comment by maxwai** [comment]
@schitcrafter this should now the fixed. We now ignore synchronize action on pull request where the branch is in the main repo

**Comment by maxwai** [comment]
nvm doesn't work yet

**Comment by maxwai** [comment]
Ok, the conditional run of check_code works, but I noticed that pushing docker container on from a pull request of a fork doesn't work because the secrets are not shared

**Comment by maxwai** [comment]
Ok, changed the docker build action to only run on push to main since the secrets can't / shouldn't be shared on pull requests of forks since this could lead to secrets being leaked.

**Comment by schitcrafter** [comment]
> Ok, changed the docker build action to only run on push to main since the secrets can't / shouldn't be shared on pull requests of forks since this could lead to secrets being leaked.

Pretty sure this isn't a concern: https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
Look under the segment starting with "Due to the dangers", it clearly states that with "pull_request", only pr's from branches on this repo will have access to secrets. Although, I still don't see why we would want to build a docker container for every pull request - maybe to test the changes in a production environment?

I'll not merge this right now because of the discussion still going on

File: .github/workflows/check-code.yml
Comment (schitcrafter) [inline .github/workflows/check-code.yml:1]: Pretty sure this could be done better, same effect could be achieved with push only on main and all pr's, if all branches that need CI/CD have a pr associated with them

**Comment by maxwai** [comment]
> Although, I still don't see why we would want to build a docker container for every pull request - maybe to test the changes in a production environment?

The idea was to check before merging that the container can be build and works. But I now recon a simple build without push should be fine on pull request and pushing only on push on main.

File: .github/workflows/check-code.yml
Comment (maxwai) [reply .github/workflows/check-code.yml:1]: But that assumes that every branch has a PR open, and that is not always the case. This solution works as long as the repository name doesn't change and that is not going to happen anytime soon.

**Comment by schitcrafter** [comment]
Well, I'd say that the actions would certainly work like this - don't think the solution is very good, but if it's stupid and works it's not stupid. I'll approve it

**Review by schitcrafter — APPROVED** [review]

**Pull request merged**
